webgoat-integration-tests/src/test/java/org/owasp/webgoat/IntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+package org.owasp.webgoat;
+
+import io.restassured.RestAssured;
+import io.restassured.config.RestAssuredConfig;
+import io.restassured.config.SSLConfig;
+import lombok.Getter;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.owasp.webwolf.WebWolf;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class IntegrationTest {
+
+    private static String WEBGOAT_URL = "http://localhost:8080/WebGoat/";
+    private static String WEBWOLF_URL = "http://localhost:9090/";
+
+    //This also allows to test the application with HTTPS when outside testing option is used
+    protected static RestAssuredConfig restConfig = RestAssuredConfig.newConfig().sslConfig(new SSLConfig().relaxedHTTPSValidation());
+
+    @Getter
+    private String webGoatCookie;
+    @Getter
+    private String webWolfCookie;
+    @Getter
+    private String webgoatUser = UUID.randomUUID().toString();
+
+    private static boolean started = false;
+
+    @BeforeClass
+    public static void beforeAll() {
+        if (!started) {
+            started = true;
+            if (!areAlreadyRunning()) {
+                SpringApplicationBuilder wgs = new SpringApplicationBuilder(StartWebGoat.class)
+                        .properties(Map.of("spring.config.name", "application-webgoat"));
+                wgs.run();
+                SpringApplicationBuilder wws = new SpringApplicationBuilder(WebWolf.class)
+                        .properties(Map.of("spring.config.name", "application-webwolf"));
+                wws.run();
+            }
+        }
+    }
+
+    private static boolean areAlreadyRunning() {
```

Comment:
Maybe make the test check on WebGoat and WebWolf separately? Or only start WebWolf for tests that involve WebWolf

Reply:
Removed the check for WebWolf if WebGoat is running it should be fine. The containers are started once and WW starts very quick so opted for starting is anyway when no container is already running.

The use cases will most probably: 1) you are writing a test/debugging it so you probably have WG already running locally with only checking WG there is no need to run WW locally as well.
2) Integration tests run during a build, since we have 1 testcase with WW the container has to started anyway